### PR TITLE
feat: add qrcode rendering

### PR DESCRIPTION
**Pull request by rjchow**
- added a popover for qrcode using react-bootstrap
- qrcode button is hidden if no url is located at { data: { links: self: href: "xx" } } }
- qrcode button is shown otherwise
- qr code format is simply a url, can use format at https://github.com/Open-Attestation/adr/blob/master/universal_actions.md
- can use ebl-with-qr-code.json in test fixtures to test
- has storybook story showing how it looks like

### CODE REVIEW

File: src/test/qrcode.spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Selector } from "testcafe";
+
+fixture("Token Document Rendering").page`http://localhost:3000`;
```
Comment (simboonlong) [inline src/test/qrcode.spec.js]: QRcode Rendering?

File: src/test/qrcode.spec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  await DocumentStatus.with({ visibilityCheck: true })();
+
+  const qrcodeButtonElement = await Selector("button").withAttribute("aria-label", "document-utility-qr-button");
+  await t.click(qrcodeButtonElement);
+  await validateTextContent(t, IssuedByDomainName, ["TRADETRUST.IO"]);
+
+  await t.switchToIframe(IframeBlock);
+
+  await validateTextContent(t, SampleTemplate, ["BILL OF LADING FOR OCEAN TRANSPORT OR MULTIMODAL TRANSPORT"]);
```
Comment (simboonlong) [inline src/test/qrcode.spec.js]: i think after click on qrCode button, maybe check if the qrCode div exists? else this test doesn't seem to check anything other than text rendered correctly from ebl (which is not the case we wanna test?)

File: src/components/DocumentUtility/DocumentUtility.tsx [hunk 3]
```diff
@@ -14,21 +16,42 @@ interface DocumentUtilityProps {
...
         <div className="row no-gutters">
           <div className="col-auto ml-auto">
-            <ButtonIconWhiteBlue onClick={() => window.print()}>
+            <OverlayTrigger trigger="click" placement="bottom-end" overlay={qrCodePopover(qrcodeUrl)}>
+              <ButtonIconWhiteBlue aria-label="document-utility-qr-button" hidden={!qrcodeUrl}>
+                <SvgIcon strokeWidth="0.5" fill="currentColor">
+                  <SvgIconQRCode />
+                </SvgIcon>
+              </ButtonIconWhiteBlue>
+            </OverlayTrigger>
```
Comment (simboonlong) [inline src/components/DocumentUtility/DocumentUtility.tsx]:
```suggestion
            {qrcodeUrl && (
              <OverlayTrigger trigger="click" placement="bottom-end" overlay={qrCodePopover(qrcodeUrl)}>
                <ButtonIconWhiteBlue aria-label="document-utility-qr-button">
                  <SvgIcon strokeWidth="0.5" fill="currentColor">
                    <SvgIconQRCode />
                  </SvgIcon>
                </ButtonIconWhiteBlue>
              </OverlayTrigger>
            )}
```

how about this? then maybe no need hidden props for button too already

File: src/components/UI/Button/Button.tsx [hunk 4]
```diff
@@ -7,6 +7,7 @@ interface ButtonProps {
   children?: React.ReactNode;
   className?: string;
   disabled?: boolean;
+  hidden?: boolean;
```
Comment (simboonlong) [inline src/components/UI/Button/Button.tsx]: maybe no need? comment above

File: src/test/qrcode.spec.js (hunk 2 above)
Comment (simboonlong) [reply src/test/qrcode.spec.js]: actually it does, qr button wouldn't even exists in the first place to be clicked 🤔. maybe can ignore my comment

File: src/components/UI/Button/Button.tsx (hunk 4 above)
Comment (rjchow) [reply src/components/UI/Button/Button.tsx]: fixed

File: src/components/DocumentUtility/DocumentUtility.tsx (hunk 3 above)
Comment (rjchow) [reply src/components/DocumentUtility/DocumentUtility.tsx]: fixed :smile:

File: src/test/qrcode.spec.js (hunk 1 above)
Comment (rjchow) [reply src/test/qrcode.spec.js]: fixed

**Pull request merged**